### DISCUSSION TITLE
No streaming interface can not support concurrency for  two fastapi servers

### DISCUSSION THREAD

**Discussion by yjjiang11** [fastapi/fastapi#10993] [Questions]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content
I set up two fastapi server as follows.
![ff486bd749b1cd3bd95906a52c8620b](https://github.com/tiangolo/fastapi/assets/34344716/95c8b122-179f-41ef-8a77-6e4de52d85e2)

But I find that streaming interfaceface can support concurrency, but no streaming interface does not support. 
My code streaming interface follows:https://github.com/lm-sys/FastChat/blob/e86e70d0b48c3d6e2223de96b20e377315dbf73e/fastchat/serve/vllm_worker.py#L182
no streaming interface follows:https://github.com/lm-sys/FastChat/blob/e86e70d0b48c3d6e2223de96b20e377315dbf73e/fastchat/serve/vllm_worker.py#L192

**Comment by jgould22** [comment]
What happens if you remove the async  code in you `api_generate_stream` function and use only a def for that method?

**Reply by yjjiang11** [reply]
api_generate_stream still supports concurrency

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
